### ISSUE TITLE
Hello, there is some confusion about the code.

### ISSUE THREAD

**Issue by KyrieXDL** [pss1207/sleep_apnea#7]
Thank you for your code, it did help me a lot. But I still have some questions about it. I don't understand why you do interpolation, is it because you can get more samples data and for the better training?

**Comment by pss1207** [comment]
@KyrieXDL Hi, The RR intervals have not a uniform sampling rate. RR interval is a difference between two QRS complexes (QRS[n] - QRS[n-1]). The time index of this RR interval[n] is the center of two QRS complexes (QRS[n-1] + QRS[n]) / 2. And, the next RR interval[n+1] has the time index (QRS[n] + QRS[n+1]) / 2. This means that the time index is changing. So, the frequency domain analysis of HRV uses interpolation and the sampling rate is usually 4 or 8 Hz. You can find a lot of papers by searching "frequency domain HRV". I thought that the fixed sample rate is better because the input data of the network is fixed. Also, we cannot reflect varying sampling rates for simple CNN or LSTM.

**Comment by KyrieXDL** [comment]
> @KyrieXDL Hi, The RR intervals have not a uniform sampling rate. RR interval is a difference between two QRS complexes (QRS[n] - QRS[n-1]). The time index of this RR interval[n] is the center of two QRS complexes (QRS[n-1] + QRS[n]) / 2. And, the next RR interval[n+1] has the time index (QRS[n] + QRS[n+1]) / 2. This means that the time index is changing. So, the frequency domain analysis of HRV uses interpolation and the sampling rate is usually 4 or 8 Hz. You can find a lot of papers by searching "frequency domain HRV". I thought that the fixed sample rate is better because the input data of the network is fixed. Also, we cannot reflect varying sampling rates for simple CNN or LSTM.

Thank you for your reply! You have got the RR intervals of the raw data before interpolation. But as you said, the RR intervals have not a uniform sampling rate, so you used the interpolation and sampling with a fixed rate. Do I understand right? But after sampling with 4 HZ, the number of RR intervals has increased, is this right? I'll spend more time to learn it.

**Comment by pss1207** [comment]
@KyrieXDL Yes, that's correct. The sampling rate of RR intervals should be selected based on the minimum RR interval. The maximum heart rate is about 300 BPM which is 5 Hz. However, HRV only considers normal sinus rhythm. This means 120 BPM (2 Hz) is enough for HRV applications. With the Nyquist sampling theorem, it's 4 Hz (2 Hz x 2).

**Comment by KyrieXDL** [comment]
> @KyrieXDL Yes, that's correct. The sampling rate of RR intervals should be selected based on the minimum RR interval. The maximum heart rate is about 300 BPM which is 5 Hz. However, HRV only considers normal sinus rhythm. This means 120 BPM (2 Hz) is enough for HRV applications. With the Nyquist sampling theorem, it's 4 Hz (2 Hz x 2).

Thank you so much for your reply!
